fcrepo-search-api/src/main/java/org/fcrepo/search/api/SearchIndex.java

```diff
@@ -17,11 +17,31 @@
  */
 package org.fcrepo.search.api;
 
+import org.fcrepo.kernel.api.identifiers.FedoraId;
+
+import java.time.Instant;
+
 /**
- * An interface defining search operations
+ * An interface defining search index management operations
+ *
  * @author dbernstein
  */
-public interface SearchService {
+public interface SearchIndex {
+
+    /**
+     * @param fedoraId
+     * @param created
+     * @param modified
+     * @param size
+     * @param mimetype
+     */
+    void addUpdateIndex(final FedoraId fedoraId, final Instant created, final Instant modified,
```

Comment:
Does this module have access to the `ResourceHeaders` interface? If so, it might be nice to have an api that just took `FedoraId` and `ResourceHeaders` rather than having to pass each value individually.

Reply:
Yeah - that's valid.  I wasn't sure if we wanted to couple the interface too tightly here.

Reply:
I'm inclined to leave it as is - perhaps @awoods  may want to weigh in.

Reply:
Are these the only fields that this method is going to accept, or are you planning on indexing more ResourceHeaders fields in the future as you build search out? If you're not planning on adding more, I'm okay with it. Otherwise, I think it would be better to have a different object in this method instead of an ever growing parameter list.

Reply:
It appears that this PR removed all the other fields, so it is now FEDORA_ID, MODIFIED or CREATED? Or is that not part of this?

Reply:
So we will be adding fields  (size and mimetype) to this list.   I suppose we could have ResourceHeaders passed in.  I'm not sure why I was initially resistant to it.  

@whikloj :  the other fields will be added back in when they are actually supported by search.  Yes I did remove them to keep things tidy until we actually need them.

Reply:
I will use ResourceHeaders.